src/test/scala/ch/epfl/pop/tests/MessageCreationUtils.scala

```diff
@@ -6,20 +6,22 @@ import spray.json.enrichAny
 import ch.epfl.pop.json.JsonCommunicationProtocol.MessageContentDataFormat
 import scorex.crypto.signatures.{Curve25519, PrivateKey, PublicKey}
 
+import java.nio.charset.StandardCharsets
 import java.security.MessageDigest
 import java.util.Base64
 
 object MessageCreationUtils {
 
   def b64Encode(b: Array[Byte]): Array[Byte] = Base64.getEncoder.encode(b)
 
+  def b64EncodeToString(b: Array[Byte]): String = Base64.getEncoder.encodeToString(b)
+
   def getMessageParams(data: MessageContentData, pk: PublicKey, sk: PrivateKey, channel: ChannelName): MessageParameters = {
     val encodedData = b64Encode(data.toJson.compactPrint.getBytes).map(_.toChar).mkString
     val signature = Curve25519.sign(sk, encodedData.getBytes())
     val md = MessageDigest.getInstance("SHA-256")
-    md.update(encodedData.getBytes())
-    md.update(signature)
-    val messageId = md.digest()
+    val id = "[\"" + encodedData + "\",\"" + b64EncodeToString(signature) + "\"]"
```

Comment:
since you already have hashing logic in crypto/Hash.scala, supporting a variable number of parameters and what not, why repeat yourself?

The concatenation and hashing is somewhat error prone, you might just want to have a generic function with a variable number of arguments, returning the hash for just 1 parameter, or the proper concatenation for 2+ parameters.

This way, you only have 1 function to test and it'll be correct everywhere. Plus, if it's tested and known to be correct, you could even rely on it in the tests such as PubSubTest.scala. 

Furthermore, if the hashing function or the concatenation logic needs to change tomorrow, you'll only have to change 1 place in the code.